systemtest/src/test/java/io/strimzi/systemtest/specific/RackAwarenessIsolatedST.java

```diff
@@ -0,0 +1,421 @@
+import io.strimzi.systemtest.annotations.IsolatedTest;
+import static io.strimzi.systemtest.k8s.Events.Created;
+import static io.strimzi.systemtest.k8s.Events.Pulled;
+import static io.strimzi.systemtest.k8s.Events.Scheduled;
+import static io.strimzi.systemtest.k8s.Events.Started;
+import io.strimzi.systemtest.kafkaclients.internalClients.KafkaClients;
+import io.strimzi.systemtest.kafkaclients.internalClients.KafkaClientsBuilder;
+import static io.strimzi.systemtest.matchers.Matchers.hasAllOfReasons;
+import io.strimzi.systemtest.resources.ResourceManager;
+import static io.strimzi.systemtest.resources.ResourceManager.cmdKubeClient;
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+import io.strimzi.systemtest.resources.crd.KafkaConnectResource;
+import io.strimzi.systemtest.resources.kubernetes.NetworkPolicyResource;
+import io.strimzi.systemtest.resources.operator.SetupClusterOperator;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaClientsTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaConnectTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaMirrorMaker2Templates;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaTopicTemplates;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaConnectUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.DeploymentUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import io.strimzi.systemtest.utils.specific.ScraperUtils;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+@Tag(REGRESSION)
+@Tag(CONNECT)
+@Tag(CONNECT_COMPONENTS)
+@Tag(INTERNAL_CLIENTS_USED)
+@IsolatedSuite
+class RackAwarenessIsolatedST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(RackAwarenessIsolatedST.class);
+
+    @IsolatedTest
+    void testKafkaRackAwareness(ExtensionContext extensionContext) {
+        assumeFalse(Environment.isNamespaceRbacScope());
+
+        TestStorage storage = new TestStorage(extensionContext);
+        String namespace = storage.getNamespaceName();
+        String clusterName = storage.getClusterName();
+        String topologyKey = "kubernetes.io/hostname";
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 1, 1)
```

Comment:
Can you elaborate more on this? We have isolated tests here, so I think you are referring to a possible switch to parallel tests, which I'm actually working on and it seems to work fine.

Reply:
Why it should be connected to parallel tests? When you use 1 Kafka and 1 ZK, the cluster is not stable - even in the warnings you can see that it's not recommended to use 1 Kafka and 1 ZK - rolling updates, message exchange, different operations.

What exactly is problem to set the 3/3 replicas?

Reply:
You talked about race conditions specifically, this is what I'm interested about. I know it's not recommended, but here I'm primarily testing the configuration and it's faster with just one replica. Switching to parallel execution is even faster (I did some changes to allow complete namespace isolation).

Reply:
Yeah, it's obviously faster. The thing is that when you are doing some additional checks, they can sometimes fail, because the cluster wasn't stable enough - that's what I mean when I'm saying _race conditions_.

Also, you can use one Kafka with 3/3 in `BeforeAll` as a shared cluster and then, in each test, deploy different resources (and additional Kafka for MM2 test)